### PR TITLE
Updates for Kubeflow 0.4.1

### PR DESCRIPTION
**Pull request by sozercan**
updates for kubeflow 0.4.1 and kfctl.sh

### CODE REVIEW

File: 2-kubernetes/README.md [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
 
 In this module you will learn:
-* The basic concepts of Kubernetes
-* How to create a Kubernetes cluster on Azure
 
-> *Important* : Kubernetes is very often abbreviated to **K8s**. This is the name we are going to use in this workshop.
+- The basic concepts of Kubernetes
+- How to create a Kubernetes cluster on Azure
+
+> _Important_ : Kubernetes is very often abbreviated to **K8s**. This is the name we are going to use in this workshop.
```
Comment (ritazh) [inline 2-kubernetes/README.md]: s/workshop/lab

File: 2-kubernetes/README.md [hunk 2]
```diff
@@ -129,6 +138,24 @@ az aks get-credentials --name <NAME> --resource-group <RG>
...
+```
+
+For Kubernetes 1.11 and above:
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/nvidia/k8s-device-plugin/v1.11/nvidia-device-plugin.yml
+```
+
+For AKS Engine, NVIDIA Device Plugin will automatically installed with N-Series GPU clusters.
+
```
Comment (ritazh) [inline 2-kubernetes/README.md]: Should we set a `KUBERNETES_VERSION` variable then use that in the url?

File: 4-kubeflow/README.md [hunk 3]
```diff
@@ -16,59 +16,49 @@ From [Kubeflow](https://github.com/kubeflow/kubeflow)'s own documetation:
...
+- [JupyterHub](https://jupyterhub.readthedocs.io/en/latest/), which allows user to request an instance of a Jupyter Notebook server dedicated to them.
+- One or multiple training controllers. These are component that simplifies and manages the deployment of training jobs. For the purpose of this lab we are only going to deploy a training controller for TensorFlow jobs. However the Kubeflow community has started working on controllers for PyTorch and Caffe2 as well.
+- A serving component that will help you serve predictions with your models.
 
 For more general info on Kubeflow, head to the repo's [README](https://github.com/kubeflow/kubeflow/blob/master/README.md).
 
 ### Deploying Kubeflow
 
-Kubeflow uses [`ksonnet`](https://github.com/ksonnet/ksonnet) templates as a way to package and deploy the different components.  
+Kubeflow uses [`ksonnet`](https://github.com/ksonnet/ksonnet) templates as a way to package and deploy the different components.
```
Comment (ritazh) [inline 4-kubeflow/README.md]: Should we create a `Deploying Dependancies` section before the `Deploying Kubeflow` section for stuff like `ks`, `argo`, `kubectl`, `az` etc.

File: 4-kubeflow/README.md [hunk 4]
```diff
@@ -16,59 +16,49 @@ From [Kubeflow](https://github.com/kubeflow/kubeflow)'s own documetation:
...
+```bash
+KUBEFLOW_SRC=kubeflow
+
+mkdir ${KUBEFLOW_SRC}
+cd ${KUBEFLOW_SRC}
+
+export KUBEFLOW_TAG=v0.4.1
 
-> ksonnet simplifies defining an application configuration, updating the configuration over time, and specializing it for different clusters and environments. 
+curl https://raw.githubusercontent.com/kubeflow/kubeflow/${KUBEFLOW_TAG}/scripts/download.sh | bash
```
Comment (ritazh) [inline 4-kubeflow/README.md]: 👍 I should update my repo to just use this lol.

File: 5-jupyterhub/README.md [hunk 5]
```diff
@@ -31,33 +33,32 @@ docker run -it -p 8888:8888 tensorflow/tensorflow
...
+jupyter-lb                               ClusterIP   10.0.191.68    <none>        80/TCP              132m
 ```
 
 To connect to your JupyterHub locally:
 
-```
-PODNAME=`kubectl get pods --namespace=${NAMESPACE} --selector="app=tf-hub" --output=template --template="{{with index .items 0}}{{.metadata.name}}{{end}}"`
-kubectl port-forward --namespace=${NAMESPACE} $PODNAME 8000:8000
+```bash
+kubectl port-forward svc/ambassador -n ${NAMESPACE} 8080:80
```
Comment (ritazh) [inline 5-jupyterhub/README.md]: To connect to the Kubeflow dashboard locally:

Then navigate to JupyterHub: http://localhost:8080/hub

File: 2-kubernetes/README.md [hunk 6]
```diff
@@ -146,33 +174,35 @@ aks-nodepool1-42640332-2   Ready     agent     1h        v1.11.1
...
 
 [...]
 Capacity:
  nvidia.com/gpu:     1
 [...]
- ```
+```
 
-> Note: In some scenarios, you might not see GPU resources under Capacity. To resolve this, you must install a daemonset as described in the troubleshooting section here: https://docs.microsoft.com/en-us/azure/aks/gpu-cluster 
+> Note: In some scenarios, you might not see GPU resources under Capacity. To resolve this, you must install a daemonset as described in the troubleshooting section here: https://docs.microsoft.com/en-us/azure/aks/gpu-cluster
```
Comment (ritazh) [inline 2-kubernetes/README.md]: Can we also add `kubectl apply -f https://raw.githubusercontent.com/nvidia/k8s-device-plugin/v1.11/nvidia-device-plugin.yml` here?

File: 2-kubernetes/README.md (hunk 2 above)
Comment (sozercan) [reply 2-kubernetes/README.md]: its only `v1.10` and `v1.11`

File: 4-kubeflow/README.md (hunk 3 above)
Comment (sozercan) [reply 4-kubeflow/README.md]: it can be a follow up pr to this

**Pull request merged**